src/core/grid_based_algorithms/lbgpu_cuda.cu

```diff
@@ -1418,9 +1418,11 @@ velocity_interpolation(LB_nodes_gpu n_a, float *particle_position,
 
   int cnt = 0;
   float3 interpolated_u{0.0f, 0.0f, 0.0f};
-#pragma unroll
+#pragma unroll 1
```

Comment:
can't you just remove the ```pragma``` directive here?

> If the ICE evaluates to 1, the compiler will not unroll the loop

from https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#pragma-unroll

Or do you want to force the compiler not to unroll here?

Reply:
I wanted to force it to not unroll, just to be on the safe side. During my (rather limited) testing, I noticed that the compilers (nvcc, hipcc, clang) sometimes do some automatic unrolling, even on this long-bodied loop.